src/VisualStudio/Core/Def/Implementation/Remote/VisualStudioWorkspaceServiceHubConnector.cs

```diff
@@ -29,7 +29,7 @@ internal sealed class VisualStudioWorkspaceServiceHubConnector : IEventListener<
         private GlobalNotificationRemoteDeliveryService? _globalNotificationDelivery;
         private Task<RemoteHostClient?>? _remoteClientInitializationTask;
         private SolutionChecksumUpdater? _checksumUpdater;
-        private CancellationTokenSource _disposalCancellationSource;
+        private readonly CancellationTokenSource _disposalCancellationSource;
```

Comment:
this feels risky as a struct.  i honestly don't see how this could be safe.  if someone calls `Cancel` on this... then it will mutate a copy, not the actual cancellation source, right?

Reply:
Should we update the analyzer for this?